docs/guides/plugins.md

```diff
@@ -1,58 +1,345 @@
-# Plugins
+# Video.js Plugins
 
-If you've built something cool with Video.js, you can easily share it with the rest of the world by creating a plugin. Although, you can roll your own, you can also use [generator-videojs-plugin](https://github.com/dmlap/generator-videojs-plugin), a [Yeoman](http://yeoman.io) generator that provides scaffolding for video.js plugins including:
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 
-* [Grunt](http://gruntjs.com) for build management
-* [npm](https://www.npmjs.org) for dependency management
-* [QUnit](http://qunitjs.com) for testing
 
-## Step 1: Write Some Javascript
+- [Writing a Basic Plugin](#writing-a-basic-plugin)
+  - [Write a JavaScript Function](#write-a-javascript-function)
+  - [Register a Basic Plugin](#register-a-basic-plugin)
+- [Writing a Advanced Plugin](#writing-a-advanced-plugin)
+  - [Write a JavaScript Class/Constructor](#write-a-javascript-classconstructor)
```

Comment:
If the new doc tooling was merged into master, it might be better to switch over to that style of TOC if we can.